planutils/package_installation.py

```diff
@@ -0,0 +1,118 @@
+
+import json, os, glob, subprocess
+from collections import defaultdict
+
+from planutils import settings
+
+PACKAGES = {}
+
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+
+for conf_file in glob.glob(os.path.join(CUR_DIR, 'packages', '*')):
+    base = os.path.basename(conf_file)
+    if base not in ['README.md', 'TEMPLATE']:
+        with open(os.path.join(conf_file, 'manifest.json'), 'r') as f:
+            config = json.load(f)
+        assert base not in PACKAGES, "Error: Duplicate package config -- %s" % base
+        PACKAGES[base] = config
+        PACKAGES[base]['runnable'] = os.path.exists(os.path.join(conf_file, 'run'))
+
+
+def check_installed(target):
+    return target in settings.load()['installed']
+
+
+def uninstall(target):
+
+    if target not in PACKAGES:
+        print("Error: Package not found -- %s" % target)
+        return
+
+    if not check_installed(target):
+        print("%s isn't installed." % target)
+    else:
+        s = settings.load()
+        # map a package to all those that depend on it
+        dependency_mapping = defaultdict(set)
+        for p in s['installed']:
+            for dep in PACKAGES[p]['dependencies']:
+                dependency_mapping[dep].add(p)
+
+        # compute all the packages that will be removed
+        to_check = [target]
+        to_remove = set()
+        while to_check:
+            check = to_check.pop(0)
+            to_remove.add(check)
+            to_check.extend(list(dependency_mapping[check]))
+
+        print("\nAbout to remove the following packages: %s" % ', '.join(to_remove))
+        if input("  Proceed? [y/N] ").lower() in ['y', 'yes']:
+            for package in to_remove:
+                print ("Uninstalling %s..." % package)
+                subprocess.call('./uninstall', cwd=os.path.join(CUR_DIR, 'packages', package))
```

Comment:
Ya, not sure...if it really mucks up, then maybe we just need a better written uninstall script. We're kind of trusting the packages to deal with things correctly, since technically anything can be thrown into the scripts.

Skipping for now...